c/tests/test_minimal_cpp.cpp

```diff
@@ -78,12 +80,122 @@ test_table_basics()
     tsk_table_collection_free(&tables);
 }
 
+/* A definition of sort_edges that uses C++ std::sort and inlining of the
+ * comparison function to achieve significantly better performance than
+ * the builtin method in tskit.
+ */
+int
+sort_edges(tsk_table_sorter_t *sorter, tsk_size_t start)
+{
+    struct _edge {
+        double left, right;
+        tsk_id_t parent, child;
+
+        _edge(double l, double r, tsk_id_t p, tsk_id_t c)
+            : left{ l }, right{ r }, parent{ p }, child{ c }
+        {
+        }
+    };
+    tsk_edge_table_t *edges = &sorter->tables->edges;
+    const double *node_time = sorter->tables->nodes.time;
+    std::vector<_edge> sorted_edges;
+    size_t num_edges = edges->num_rows;
+    size_t j;
+
+    /* This is the comparison function.  We cannot define an
+     * operator < for _edge because we need to bind the node times
+     * so we have to use a functional method. This is a copy of the cmp
+     * from fwdpp.  Only difference is the final time comparison
+     * (fwdpp table times to forwards). */
+    const auto cmp = [&node_time](const _edge &lhs, const _edge &rhs) {
+        auto tl = node_time[lhs.parent];
+        auto tr = node_time[rhs.parent];
+        if (tl == tr) {
+            if (lhs.parent == rhs.parent) {
+                if (lhs.child == rhs.child) {
+                    return lhs.left < rhs.left;
+                }
+                return lhs.child < rhs.child;
+            }
+            return lhs.parent < rhs.parent;
+        }
+        return tl < tr;
+    };
+
+    assert(start == 0);
+    /* Let's not bother with metadata */
+    assert(edges->metadata_length == 0);
+
+    sorted_edges.reserve(num_edges);
+    for (j = 0; j < num_edges; j++) {
+        sorted_edges.emplace_back(
```

Comment:
This is the most likely place to throw an exception--you can get an "out of memory" here.  The call to std::sort is quite unlikely to throw.  Even if the cmp function is really daft and tries to compare "parent + 2", the behavior is undefined like in C: return random gibberish or segfault.